core/chains/evm/txmgr/orm.go

```diff
@@ -943,3 +959,275 @@ func (o *orm) SaveReplacementInProgressAttempt(oldAttempt EthTxAttempt, replacem
+		return errors.New("expected error field to be set")
+	}
+
+	etx.Nonce = nil
+	etx.State = EthTxFatalError
+
+	return qq.Transaction(func(tx pg.Queryer) error {
+		if _, err := tx.Exec(`DELETE FROM eth_tx_attempts WHERE eth_tx_id = $1`, etx.ID); err != nil {
+			return errors.Wrapf(err, "saveFatallyErroredTransaction failed to delete eth_tx_attempt with eth_tx.ID %v", etx.ID)
+		}
+		return errors.Wrap(tx.Get(etx, `UPDATE eth_txes SET state=$1, error=$2, broadcast_at=NULL, initial_broadcast_at=NULL, nonce=NULL WHERE id=$3 RETURNING *`, etx.State, etx.Error, etx.ID), "saveFatallyErroredTransaction failed to save eth_tx")
+	})
+}
+
+type QueryerFunc func(tx pg.Queryer) error
+
+// Updates eth attempt from in_progress to broadcast. Also updates the eth tx to unconfirmed.
+// Before it updates both tables though it increments the next nonce from the keystore
+// One of the more complicated signatures. We have to accept variable pg.QOpt and QueryerFunc arguments
+func (o *orm) UpdateEthTxAttemptInProgressToBroadcast(etx *EthTx, attempt EthTxAttempt, NewAttemptState EthTxAttemptState, incrNextNonceCallback QueryerFunc, args ...any) error {
+	qopts := []pg.QOpt{}
+	afterSaveCallbacks := []QueryerFunc{}
+	for _, value := range args {
+		switch v := value.(type) {
+		case QueryerFunc:
+			afterSaveCallbacks = append(afterSaveCallbacks, v)
+		case pg.QOpt:
+			qopts = append(qopts, v)
+		default:
+			return fmt.Errorf("unable to accept arg of type %T", value)
+		}
+	}
+
+	qq := o.q.WithOpts(qopts...)
+
+	if etx.BroadcastAt == nil {
+		return errors.New("unconfirmed transaction must have broadcast_at time")
+	}
+	if etx.InitialBroadcastAt == nil {
+		return errors.New("unconfirmed transaction must have initial_broadcast_at time")
+	}
+	if etx.State != EthTxInProgress {
+		return errors.Errorf("can only transition to unconfirmed from in_progress, transaction is currently %s", etx.State)
+	}
+	if attempt.State != EthTxAttemptInProgress {
+		return errors.New("attempt must be in in_progress state")
+	}
+	if NewAttemptState != EthTxAttemptBroadcast {
+		return errors.Errorf("new attempt state must be broadcast, got: %s", NewAttemptState)
+	}
+	etx.State = EthTxUnconfirmed
+	attempt.State = NewAttemptState
+	return qq.Transaction(func(tx pg.Queryer) error {
+		if err := incrNextNonceCallback(tx); err != nil {
+			return errors.Wrap(err, "SaveEthTxAttempt failed on incrNextNonceCallback")
+		}
+		if err := tx.Get(etx, `UPDATE eth_txes SET state=$1, error=$2, broadcast_at=$3, initial_broadcast_at=$4 WHERE id = $5 RETURNING *`, etx.State, etx.Error, etx.BroadcastAt, etx.InitialBroadcastAt, etx.ID); err != nil {
+			return errors.Wrap(err, "SaveEthTxAttempt failed to save eth_tx")
+		}
+		if err := tx.Get(&attempt, `UPDATE eth_tx_attempts SET state = $1 WHERE id = $2 RETURNING *`, attempt.State, attempt.ID); err != nil {
+			return errors.Wrap(err, "SaveEthTxAttempt failed to save eth_tx_attempt")
+		}
+		for _, f := range afterSaveCallbacks {
```

Comment:
we don't actually pass any callbacks in for any use cases. can we remove this parameter? or is this for future proofing

Reply:
Interesting, this must have been used at one point. It's possible it's no longer used - if tests pass after removing it then 🪓 it

Reply:
Did you check the git history for when this was added and why?

Reply:
Seems like once upon a time it was used for [rebroadcasting a tx ](https://github.com/smartcontractkit/chainlink/blame/72206c73f89d3406137e7181d15ab3fe5e944244/core/services/bulletprooftxmanager/eth_broadcaster.go#L324)whos nonce was used by an external wallet. . It was removed [here ](https://github.com/smartcontractkit/chainlink/commit/44d5e4a47b2bbbd4b94c87a57c3404b52d3d6b0a#diff-acc3a29fd4e9e3038da9546de1b38f7a8309d0ebbb4d284497b1fedb8a3ba06eL328)

Does not seem to be in use anymore so I'll remove it